index.bs

```diff
@@ -40,7 +72,33 @@ The {{AudioSession}} is the main interface for this API, which is accessed throu
   };
 </pre>
 
-## Audio session states ## {#audio-session-types}
+To create an {{AudioSession}} object in |realm|, run the following steps:
```

Comment:
How about moving "Extensions to the Navigator Interface" to this section to improve the clarity of the specification? First, we could establish that the AudioSession is a singleton per window, followed by the steps for creating an AudioSession.

Additionally, I would like to understand the purpose of `isTypeBeingApplied`. For instance, if we first set the type to A and then immediately change it to B, the current specification indicates that type B won't be applied correctly because `isTypeBeingApplied` will remain true for type A if the queued task hasn't executed. What is the rationale behind this mechanism? Shouldn't we allow a new type to be set for an `AudioSession` at all times?

Reply:
I am fine reshuffling a bit the spec to make it clearer. I'd like to minimize the rebasing of this PR though.

Wrt `isTypeBeingApplied`, the principle is that we only update the `audio session` type once per event loop.
The `AudioSession` object's .type can be changed many times, but only one task will be run to update its `audio session` type, with the last type set on the `AudioSession` object.

Reply:
> I am fine reshuffling a bit the spec to make it clearer. I'd like to minimize the rebasing of this PR though.

Sounds good, thanks!

> I am fine reshuffling a bit the spec to make it clearer. I'd like to minimize the rebasing of this PR though.

It sounds like it's similar with updating a MSE source buffer? If so, should we add a [updating](https://w3c.github.io/media-source/#dom-sourcebuffer-updating) to indicate the update is still ongoing? In addition, is there a particular reason we only want to update type once per event loop?

Reply:
Let's keep this one open until we discuss the exact algorithms.